File: src/com/trapps/jvsk/java/StringTemplateExample.java
package com.trapps.jvsk.java;

/**
 * @author Akash Patra
 */
public class StringTemplateExample {

    /**
     * String.format
     */
    public void showStringFormat() {
        int a = 20;
        final String t1 = String.format("Amy age is %d", a);
        System.out.println(t1);

        a = 25;
        final String t2 = String.format("%s, now is %d.", t1.replace("is", "was"), a);
        System.out.println(t2);
    }
}

File: src/com/trapps/jvsk/kotlin/StringTemplateExample.kt
package com.trapps.jvsk.kotlin

/**
 * @author Akash Patra
 */
class StringTemplateExample {

    /**
     * String Template
     */
    fun showTemplate() {
        var a = 20
        //
        val t1 = "Amy age is $a"

        println(t1)

        a = 25
        val t2 = "${t1.replace("is", "was")}, now is $a."

        println(t2)
    }
}

